packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2851,6 +2842,43 @@ export class ContainerRuntime
 		}
 	}
 
+	private scheduleFlush() {
+		if (this.flushTaskExists) {
+			return;
+		}
+
+		this.flushTaskExists = true;
+		const flush = () => {
+			this.flushTaskExists = false;
+			this.flush();
+		};
+
+		switch (this.flushMode) {
+			case FlushMode.TurnBased:
+				Promise.resolve()
+					.then(() => flush())
+					.catch((error) => {
+						this.closeFn(error as GenericError);
+					});
+				break;
+			case FlushMode.Async:
+				setTimeout(() => {
+					try {
```

Comment:
There is probably a chance to reduce bundle size a bit and reuse a bit of more code:

```ts
const act = () => {
    try {
        flush();
    } catch (error) {
        this.closeFn(error as GenericError);
    }
};

Promise.resolve().then(act);
setTimeout(act, 0);

```